sympy/core/tests/test_evalf.py

```diff
@@ -473,3 +473,22 @@ def test_issue_9326():
     d2 = Dummy('d')
     e = d1 + d2
     assert e.evalf(subs = {d1: 1, d2: 2}) == 3
+
+def test_issue_10323_ceiling():
+    assert ceiling(Rational(2**100)+Rational(2**-100)) == 2**100+1
+    assert floor(Rational(2**100)+Rational(2**-100)) == 2**100
+    assert ceiling(Rational(2**100)-Rational(2**-100)) == 2**100
+    assert floor(Rational(2**100)-Rational(2**-100)) == 2**100-1
+
+def test_issue_10323_sqrt():
+    def asserts(x,rf,rc):
+        assert (rf == rc) == (rf**2) == x
```

Comment:
You need paren around everything after the 2nd `==`: `(rf == rc) == ((rf**2) == x)`...but these tests are overkill for the problem. The failure is in the if-condition for the extra computation in `get_integer_part`. If you find a condition that is not covered by the logic already, that would be a good test. In the PR that I submitted I put a single `sqrt` expression like those tested here.